### PR TITLE
store: "permanent" errors dont close ui

### PR DESCRIPTION
**Pull request by jazzdan**
Looks like this:

Screen Shot 2019-09-25 at 12 05 58 PM

If a permanent, now called "fatal", error happens the Upper loop stops but the UI stays up and you can still browse through it.

Nothing is displayed in the web UI, and I'm not sure if that's a feature or a bug.

### CODE REVIEW

File: internal/engine/upper.go [hunk 1]
```diff
@@ -133,6 +133,9 @@ func (u Upper) Init(ctx context.Context, action InitAction) error {
 }
 
 func upperReducerFn(ctx context.Context, state *store.EngineState, action store.Action) {
+	if state.FatalError != nil {
```
Comment (nicks) [inline internal/engine/upper.go]: do you need to be able to handle the hud.ExitAction even in the error state? The DumpEngineStateAction might be good too

**Comment by nicks** [comment]
can you also fix this to display something reasonable in the web ui? fine to do in a follow-up pr

**Comment by jazzdan** [comment]
@nicks I will do that in a separate PR after you merge #2259

File: internal/engine/upper.go (hunk 1 above)
Comment (jazzdan) [reply internal/engine/upper.go]: ah good point!

**Comment by nicks** [comment]
merged!

**Pull request merged**